packages/rfw/example/wasm/lib/main.dart

```diff
@@ -31,50 +38,50 @@ class Example extends StatefulWidget {
 }
 
 class _ExampleState extends State<Example> {
-  final Runtime _runtime = Runtime();
-  final DynamicContent _data = DynamicContent();
-  late final WasmInstance _logic;
+  final Runtime _rfwRuntime = Runtime();
+  final DynamicContent _rfwData = DynamicContent();
+
+  late final Network _network;
+  late final Wasm _wasm;
 
   @override
   void initState() {
     super.initState();
     RendererBinding.instance.deferFirstFrame();
-    _runtime.update(const LibraryName(<String>['core', 'widgets']), createCoreWidgets());
-    _loadLogic();
+    _rfwRuntime.update(
+      const LibraryName(<String>['core', 'widgets']),
+      createCoreWidgets(),
+    );
+
+    // These produce platform specific implementations for network fetching
+    // and Wasm loading and calling.
+    _network = NetworkImplementation();
+    _wasm = WasmImplementation();
+
+    _loadRfwAndWasm();
   }
 
-  late final WasmFunction _dataFetcher;
+  Future<void> _loadRfwAndWasm() async {
+    final List<int> interfaceBytes = await _network.get(interfaceUrl);
+    _rfwRuntime.update(
+      const LibraryName(<String>['main']),
+      decodeLibraryBlob(Uint8List.fromList(interfaceBytes)),
+    );
+
+    await _wasm.loadModule(logicUrl);
 
-  Future<void> _loadLogic() async {
-    final DateTime expiryDate = DateTime.now().subtract(const Duration(hours: 6));
-    final Directory home = await getApplicationSupportDirectory();
-    final File interfaceFile = File(path.join(home.path, 'cache.rfw'));
-    if (!interfaceFile.existsSync() || interfaceFile.lastModifiedSync().isBefore(expiryDate)) {
-      final HttpClientResponse client = await (await HttpClient().getUrl(Uri.parse(interfaceUrl))).close();
-      await interfaceFile.writeAsBytes(await client.expand((List<int> chunk) => chunk).toList());
-    }
-    final File logicFile = File(path.join(home.path, 'cache.wasm'));
-    if (!logicFile.existsSync() || logicFile.lastModifiedSync().isBefore(expiryDate)) {
-      final HttpClientResponse client = await (await HttpClient().getUrl(Uri.parse(logicUrl))).close();
-      await logicFile.writeAsBytes(await client.expand((List<int> chunk) => chunk).toList());
-    }
```

Comment:
i would prefer if we kept this caching logic (at least in the desktop version, i guess the web version probably gets caching for free via the browser).

Reply:
i'm not too worried about it distracting since it would be hidden in the NetworkImplementation.

Reply:
sure. Kept it